src/components/Search/index.tsx

```diff
@@ -105,20 +111,41 @@ const handleSearchResult = async (
 const Search: FC<{
   content?: string
   hasButton?: boolean
+  truncateTypeHash?: boolean
   onEditEnd?: () => void
-}> = memo(({ content, hasButton, onEditEnd }) => {
+}> = memo(({ content, hasButton, onEditEnd, truncateTypeHash }) => {
   const isMobile = useIsMobile()
   const { t } = useTranslation()
+  const searchByType = cacheService.get(SEARCH_TYPE_KEY) || 'id'
```

Comment:
This looks like a setting rather than a cache, so it should be implemented using AppSettings instead of CacheService.